src/pe/mod.rs

```diff
@@ -267,6 +269,77 @@ impl<'a> PE<'a> {
     }
 }
 
+impl<'a> ctx::TryIntoCtx<scroll::Endian> for PE<'a> {
+    type Error = error::Error;
+
+    fn try_into_ctx(self, bytes: &mut [u8], ctx: scroll::Endian) -> Result<usize, Self::Error> {
+        let mut offset = 0;
+        bytes.gwrite_with(self.header, &mut offset, ctx)?;
+        // sections table and data
+        for section in self.sections {
+            // for size size_of_raw_data
+            // if < virtual_size, pad with 0
+            // align on a boundary as per file alignement field
+            // todo: figure out section alignement restrictions wrt to architecture
```

Comment:
```
        pe.header
            .optional_header
            .unwrap()
            .windows_fields
            .section_alignment
```

This is an optional field but it's everywhere. if absent, we can probably set it to 4k